BoxBrowseSDK/BoxBrowseSDK/BOXFolderViewController.m

```diff
@@ -400,29 +400,40 @@ - (void)fetchItemsWithCompletion:(void (^)(NSArray *items))completion
     folderRequest.SDKIdentifier = BOX_BROWSE_SDK_IDENTIFIER;
     folderRequest.SDKVersion = BOX_BROWSE_SDK_VERSION;
     folderRequest.requestAllFolderFields = YES;
-    [folderRequest performRequestWithCompletion:^(BOXFolder *folder, NSError *error) {
-        if (error) {
-            [self didFailToLoadItemsWithError:error];
+
+    void (^completionBlock)(NSArray *, NSError *) = ^void(NSArray *items, NSError *error) {
+        if (error == nil) {
+            self.title = self.folder.name;
+            completion(items);
+        } else {
             self.navigationController.toolbarHidden = YES;
+            [self didFailToLoadItemsWithError:error];
+        }
+
+        if (self.tableView.visibleCells.count < 1) {
+            [self switchToEmptyStateWithError:error];
+            self.searchController.searchBar.hidden = YES;
         } else {
+            [self switchToNonEmptyView];
+            self.searchController.searchBar.hidden = NO;
+        }
+    };
+
+    __block BOOL retrievedCachedFolder = NO;
+
+    [folderRequest performRequestWithCached:^(BOXFolder *folder, NSError *error) {
```

Comment:
we want to fetch items from cache in case folder is fetched from cache, and fetch items from API in case folder is fetched from API. So there will be 2 times completionBlock gets called with either both folder+items are from cache, or both are from API. We have done similarly for shareSDK https://gitenterprise.inside-box.net/Mobile/box-ios-share-sdk-source/pull/34/files#diff-a3002c10c6929401dd029c4fabf12ecfR135

We can have input to fetchItemsInFolder to indicate fetching from cache or API only.